bgpd/bgp_route.c

```diff
@@ -7646,6 +7645,7 @@ void route_vty_out(struct vty *vty, const struct prefix *p,
 	 * attr->mp_nexthop_global_in
 	 */
 	if ((safi == SAFI_ENCAP) || (safi == SAFI_MPLS_VPN)) {
+		int len;
```

Comment:
I see that this declaration is repeated in each branch of the conditionals. Please, could you move the declaration of `len` and `buf` to the beginning of the function?

If all uses of it is just for `vty_out` or `zlog`, then we could just get rid of it and use `%pI4` or `%pI6` instead (see http://docs.frrouting.org/projects/dev-guide/en/latest/logging.html?highlight=printfrr#extensions for more information).

`vty_out` uses `printfrr` as it seems: https://github.com/FRRouting/frr/blob/master/lib/vty.c#L168

Reply:
@rzalamena dropped inet_ntop with buf and replaced using %pI4/%pI6. Thanks for remembering this for me :) Regarding the test which is removed, I don't think it's needed since we add nexthop hostname in JSON in any case (does not matter if `show-nexthop-hostname` is enabled or not).